content/300-guides/100-prisma-guides/100-prisma-client-transactions-guide.mdx

```diff
@@ -0,0 +1,773 @@
+
+```prisma
+model Team {
+  id   Int    @id @default(autoincrement())
+  name String
+  members User[] // Many team members
+}
+
+model User {
+  id    Int    @id @default(autoincrement())
+  email String @unique
+  teams Team[] // Many teams
+}
+```
+
+The most straightforward approach is to create a team, then create and attach a user to that team:
+
+```ts
+// Create a team
+await prisma.team.create({
+  data: {
+    name: 'Aurora Adventures',
+  },
+})
+
+// Create a user and assign them to the team
+await client.user.create({
+  data: {
+    email: 'alice@prisma.io',
+    team: {
+      connect: {
+        id: team.id,
+      },
+    },
+  },
+})
+```
+
+However, this code has an problem - consider the following scenario:
+
+1. Creating the team succeeds - "Aurora Adventures" is now taken
+2. Creating and connecting the user fails - the team "Aurora Adventures" exists, but has no users
+3. Going through the sign-up flow again and attempting recreate "Aurora Adventures" fails - the team already exists
+
+Creating a team and adding a user should be one atomic operation that **succeeds or fails as a whole**.
+
+To implement atomic writes in a low-level database clients, you must wrap your inserts in `BEGIN`, `COMMIT` and `ROLLBACK` queries. Prisma Client solves the problem with nested writes. The following query creates a team, creates a user, and connects the records in a single transaction:
+
+```ts
+await client.team.create({
+  data: {
+    name: 'Aurora Adventures',
+    members: {
+      create: {
+        email: 'alice@prisma.io',
+      },
+    },
+  },
+})
+```
+
+Furthermore, if an error occurs at any point, Prisma Client rolls back the entire transaction.
+
+#### Nested writes FAQs
+
+##### Why can't I use the `$transaction` API to solve the same problem?
+
+The `$transaction` API does not allow you to pass IDs between distinct operations. In the following example, `createUserOperation.id` is not available yet:
+
+```ts highlight=12;delete
+const createUserOperation = prisma.user.create({
+  data: {
+    email: 'ebony@prisma.io',
+  },
+})
+
+const createTeamOperation = prisma.team.create({
+  data: {
+    name: 'Aurora Adventures',
+    members: {
+      connect: {
+        id: createUserOperation.id, // Not possible, ID not yet available
+      },
+    },
+  },
+})
+
+await prisma.$transaction([createUserOperation, createTeamOperation])
+```
+
+##### Are cascading deletes supported by the Prisma Client nested writes?
+
+No, you cannot use nested writes to perform a cascading delete - for example, delete a team and all of its members. You can, however, use the `$transaction` API to perform a cascading delete. Deleting a team and deleting the team's members are not dependent writes - if you know the ID of the team, you can [perform both operations within a `$transaction`](#scenario-privacy-legislation).
```

Comment:
It could be worth linking to the cascade delete GitHub issue to mention that we are looking at supporting that in the schema or is that not something worth mentioning here?